lucetc/src/lib.rs

```diff
@@ -261,7 +261,7 @@ impl<T: AsLucetc> LucetcOpts for T {
 }
 
 impl Lucetc {
-    pub fn new<P: AsRef<Path>>(input: P) -> Self {
+    pub fn new(input: impl AsRef<Path>) -> Self {
```

Comment:
I like to treat PRs as an opportunity for me to learn something new about Rust and try out my vocab words.  If I understand correctly, this `impl Trait` syntax is a way to introduce a concrete type parameter bounded by the supplied trait.  So, in this case, `input` must implement `AsRef<Path>`.  Do I have that right?